apis/quay/v1/quayregistry_types.go

```diff
@@ -440,6 +447,15 @@ func EnsureDefaultComponents(ctx *quaycontext.QuayRegistryContext, quay *QuayReg
 	return nil
 }
 
+// hasAffinity checks if any anti/affinity option has been changed
+// returns true if something has changed, false otherwise
+func hasAffinity(component Component) bool {
+	overrideAffinity := component.Overrides.Affinity
+	emptyAffinity := v1.Affinity{}
+	
+	return !(overrideAffinity == emptyAffinity)
```

Comment:
`return overrideAffinity != emptyAffinity`

Reply:
well, once you convert `Affinity` to be a pointer this would become simply `component.Overrides.Affinity != nil`

Reply:
> `return overrideAffinity != emptyAffinity`

That's some next gen coding from me :/